### ISSUE TITLE
Use gyb from the Swift project for the AppKit/UIKit extensions

### ISSUE THREAD

**Issue by liscio** [neilpa/Rex#76]
The `MutableProperty` wrappers are effectively copy & pasted with a few tweaks here & there for types, associated keys, and so on. It would be wise to figure out a way to use the `gyb` (Generate Your Boilerplate) tool from the Swift project to help us overcome a lot of that (sadly necessary) boilerplate.

**Comment by neilpa** [comment]
Good idea - https://github.com/apple/swift/blob/master/utils/gyb.py

**Comment by andersio** [comment]
Implemented in ~~#142~~ #147. :)

**Comment by dmcrodrigues** [comment]
@andersio In my humble opinion, that should have been done in a separate PR and not in a branch expected to restructure/rename a lot of things and which may not be merged soon. I think this won't make our life easier because it will require a synchronization between current version targeting Swift 2.x and the new one targeting Swift 3.

**Comment by andersio** [comment]
I agree we need a branch like `RAC5` in the main repo for reviewed PRs.

But for gyb particularly, I don't see we would back port it for any Swift 2.x release, which would not be broken without it.

Moreover, it actually helps ease our development effort of RAC5 in further restructuring, like dropping prefixes or even introducing new property types specifically for them. IMO we would probably want it to land as part of the Swift 3 renaming.

**Comment by andersio** [comment]
@dmcrodrigues They are now split, and I have setup the `RAC5` as the branch for reviewed PR like the main repo.
